src/cosmosdb-preview/azext_cosmosdb_preview/custom.py

```diff
@@ -170,29 +234,76 @@ def cli_cosmosdb_managed_cassandra_datacenter_create(client,
                                                      data_center_location,
                                                      delegated_subnet_id,
                                                      node_count,
-                                                     base64_encoded_cassandra_yaml_fragment=None):
+                                                     base64_encoded_cassandra_yaml_fragment=None,
+                                                     managed_disk_customer_key_uri=None,
+                                                     backup_storage_customer_key_uri=None,
+                                                     sku=None,
+                                                     disk_sku=None,
+                                                     disk_capacity=None,
+                                                     availability_zone=None,
+                                                     server_hostname=None,
+                                                     server_port=None,
+                                                     service_user_distinguished_name=None,
+                                                     service_user_password=None,
+                                                     search_base_distinguished_name=None,
+                                                     search_filter_template=None,
+                                                     server_certificates=None):
 
     """Creates an Azure Managed Cassandra Datacenter"""
 
+    ldap_properties = AuthenticationMethodLdapProperties(
+        server_hostname=server_hostname,
+        server_port=server_port,
+        service_user_distinguished_name=service_user_distinguished_name,
+        service_user_password=service_user_password,
+        search_base_distinguished_name=search_base_distinguished_name,
+        search_filter_template=search_filter_template,
+        server_certificates=server_certificates
+    )
+
+    isAllNone = True
+    for value in vars(ldap_properties).values():
+        if value is not None:
+            isAllNone = False
+
     data_center_properties = DataCenterResourceProperties(
         data_center_location=data_center_location,
         delegated_subnet_id=delegated_subnet_id,
         node_count=node_count,
-        base64_encoded_cassandra_yaml_fragment=base64_encoded_cassandra_yaml_fragment
+        base64_encoded_cassandra_yaml_fragment=base64_encoded_cassandra_yaml_fragment,
+        sku=sku,
+        disk_sku=disk_sku,
+        disk_capacity=disk_capacity,
+        availability_zone=availability_zone,
+        managed_disk_customer_key_uri=managed_disk_customer_key_uri,
+        backup_storage_customer_key_uri=backup_storage_customer_key_uri
     )
 
+    if isAllNone is False:
+        data_center_properties.authentication_method_ldap_properties=ldap_properties
```

Comment:
When this if condition is not met, data_center_properties.authentication_method_ldap_properties is not set at all, but still the serialized payload contains data_center_properties.authentication_method_ldap_properties as {}

Instead I expect authentication_method_ldap_properties key not to be part of the payload itself.